generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/Resolver.java

```diff
@@ -146,7 +146,7 @@ public void resolveArtifact(String artifactCoord) {
    * file.
    */
   @VisibleForTesting
-  void traverseDeps(Model model, Set<String> exclusions, Rule parent) {
+  void traverseDeps(Model model, Set<String> scopes, Set<String> exclusions, Rule parent) {
```

Comment:
The `scopes` you pass into this function are the ones that you are considering worth traversing. Correct?

I see above, you are only traversing `compile` deps

Reply:
These scopes filter which dependencies we'll include from the root project. At least, that's the intent - I believe the `parent` parameter indicates the parent *dependency*, not the parent *project*, yes?

Another option would be to have a filter in [the `resolvePomDependencies` function](https://github.com/bazelbuild/migration-tooling/blob/cc88b8da996c222a3819aa006e795b68dbe983bd/generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/Resolver.java#L115-L118), before calling `traverseDeps` for the first time.

Reply:
>  I believe the parent parameter indicates the parent dependency, not the parent project, yes?

That is correct. It is the parent dependency. 
> Another option would be to have a filter in the resolvePomDependencies function, before calling traverseDeps for the first time.

I am not confident that is a better approach, at least if we do it manually. Aether has a rich set of functions for filtering and modifying the universe of dependencies. However, that is not completely integrated. This seems like a happy medium.